### DISCUSSION TITLE
plain (pdf)LaTeX Project's fail with fresh TexLive installation - here is a Solution...

### DISCUSSION THREAD

**Discussion by paule32** [doxygen/doxygen#11201] [General]
Hello,
yesterday I faced with a Problem by creating PDF Documentation with Doxygen.
Doxygen create LaTeX files before you can create a PDF file (so, Doxygen does not really create PDF files by it's own).

There are two options that you can use:
- **TexLive** (not recommend - but it is the safety variant of all)
- **MikTex** (what I would recommend)
On fresh Installation of **TexLive** you will be faced with the Problem, that the file **varwidth.sty** can not be found.
To do **not** waste your spare time, I recommend that you use MikTex.
You can download **MikText** for your Operating System [here](https://miktex.org/download).
When you finish this first step, and installed it on your local storage, you have to extend the System Environment Variable, namely **PATH**.
The PATH Variable should contain the path to the MikTex binaries directory, so you can call the EXE cutables that are shipped with MikTex Distribution.
On my local Windows 11 64-Bit Pro Developer domain, this value is:
**L:/MikTeX/miktex/bin/x64/**

You can then call:
**L:/MikTeX/miktex/bin/x64/miktex-pdflatex.exe doc.tex**

by a Windows Console Batch file.

**Note**: that you have to call miktex-pdflatex.exe multiple twice because pdflatex collect Information during the first call that are currently available at second transformation stage.

For me, I manually create the PDF with the Doxygen created *refman.tex* file.
And on a fresh run from MikTex that can take a while, too.
Don't cry on this - the second transformation will be faster.

Doxygen creates a Makefile, and a make.bat file - but these have to be modified a little bit, to fit in your local Developer domain.

**Conclusion:**
- use TexLive if you have concerns about the Security of the PDF and TEX files (because you can faced with viruses in the PDF file that can be injected with normal **LaTeX** Compilers. But don't be shy - TexLive is focused to remove this fact by strict using rules.
- TexLive is harder to installed and maintained (The consultation of an Administrator is recommend)

- use MikTex instead TexLive. **MikTex** is a special Version for Microsoft for easy install and use. But it covers not so strict rules like TexLive it does.
- try to manually convert the refman.tex within the MikTex Editor.
- MikTex is easy to install and use (maybe it is a little bit faster than TexLive - because the strict checks of TexLive)

If you have any Question or you are not sure what to do, drop a message. And we can look how we can help.

Thanks for reading.
paule32

**Comment by albert-github** [comment]
> You can then call:
> `L:/MikTeX/miktex/bin/x64/mikex-pdflatex.exe doc.tex`

This is a not a good idea as e.g the `makeidx` and `biblatex` aren't called. Doxygen provides for this purpose, on the latex directory a file `make.bat` (and a `Makefile`, for *nix) that will pickup the pdflatex etc from the path an when necessay the user can also set the (environment) variables like `LATEX_CMD` (on *nix also as arguments  of the make command).

> Doxygen creates a Makefile, and a make.bat file - but these have to be modified a little bit, to fit in your local Developer domain.

so this is not necessary, see above.
